tls/s2n_resume.c

```diff
@@ -57,7 +57,11 @@ static int s2n_tls12_serialize_resumption_state(struct s2n_connection *conn, str
     POSIX_GUARD(conn->config->wall_clock(conn->config->sys_clock_ctx, &now));
 
     /* Write the entry */
-    POSIX_GUARD(s2n_stuffer_write_uint8(to, S2N_TLS12_SERIALIZED_FORMAT_VERSION));
+    if (conn->ems_negotiated && S2N_IN_TEST) {
```

Comment:
Where is ems_negotiated set? Why does it need be separate from ems_required?

Reply:
ems_negotiated is set if both the client and the server send the EMS extension in their client hello/server hello respectively. ems_required is set when a session ticket includes the ems information, which means the resulting handshake is required to negotiate ems.